### DISCUSSION TITLE
String comparison complains about ampersands

### DISCUSSION THREAD

**Discussion by rigrig** [scandum/tintin#67] [Bugs]
Basically I'm doing something like this, but getting a warning:
```
#debug all
...
#math isExample {{http://example.com?foo=1&bar=2} == {%*example.com%*}}
    #math isExample {{http://example.com?foo=1&bar=2} == {%*example.com%*}}
#WARNING: FOUND &bar. USE &{bar} INSTEAD?
#MATH: VARIABLE {isExample} HAS BEEN SET TO {1}.
```
It's coming from [substitute.c](https://github.com/scandum/tintin/blob/e8686858ef49772e540d8457e805958c760b0f0e/src/substitute.c#L334), there are two checks, I added some debug output and the `root == NULL` one fired here.

I guess it makes sense to complain if you just type this:
```
#math test {{&a} == {}}
    #math test {{&a} == {}}
#WARNING 1: FOUND &a in '&a} == {}'. USE &{a} INSTEAD?
#MATH: VARIABLE {test} HAS BEEN SET TO {0}.
```

But my actual code is a bunch of `#if` statements that decide whether to substitute URL-like strings with clickable links, so it is triggered by `%0` containing the offending character: (thus hard to work around, unless I'm doing something wrong here?)
```
#nop URLs: regex from https://regexr.com/3e6m0;
#sub {{(http(s)?:\/\/.)?(www\.)?[-a-zA-Z0-9@:%._\+~#=]{2,256}\.[a-z]{2,6}\b([-a-zA-Z0-9@:%_\+.~#?&//=]*)}}
    {@url_link{%0}};
#nop Link to open URL: @url_link{url};
#function {url_link} {
   #if {"%0" == "%*..%*"} #return {%0}; #nop probably not an URL after all <-- WARNING;
   ...
   #return <clickable link>;
}
```

I'm running `2.02.12b`, looks like this warning was added in [`2.02.04`](https://github.com/scandum/tintin/commit/6d606547261529f26eae056c3bc9df88da417317#diff-9ec971ee6c78232ebd2d5e9f094aa1e04b9a0f976701b4036907889f6c29b289R334).

**Comment by scandum** [comment]
That's pretty annoying and I reckon that warning has outlived its usefulness. I've removed the warning in the beta.

**Reply by rigrig** [reply]
Great, thanks!
